connectors/spark/src/test/java/io/unitycatalog/spark/auth/AwsCredentialRenewalTest.java

```diff
@@ -0,0 +1,362 @@
+
+    // Initialize the schema in unity catalog server.
+    schemaOperations = new SdkSchemaOperations(createApiClient(serverConfig));
+    schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(CATALOG_NAME));
+  }
+
+  @AfterEach
+  public void afterEach() {
+    // Drop the table.
+    sql("DROP TABLE IF EXISTS %s", TABLE_NAME);
+
+    // Delete the scheme.
+    callQuietly(() -> schemaOperations.deleteSchema(SCHEMA_NAME, Optional.of(true)));
+
+    // Delete the catalog.
+    callQuietly(() -> catalogOperations.deleteCatalog(CATALOG_NAME, Optional.of(true)));
+
+    // Close the session.
+    callQuietly(() -> session.close());
+
+    super.cleanUp();
+  }
+
+  @AfterAll
+  public static void afterAll() {
+    Clock.removeManualClock(CLOCK_NAME);
+  }
+
+  public static Clock testClock() {
+    return Clock.getManualClock(CLOCK_NAME);
+  }
+
+  @Test
+  public void testFileSystemRenewal() throws Exception {
+    String location = String.format("s3://%s%s/fs", BUCKET_NAME, dataDir.getCanonicalPath());
+    Path locPath = new Path(location);
+
+    // Create the external delta path-based table.
+    S3CredFileSystem.credentialCheckEnabled = false;
+    sql("CREATE TABLE delta.`%s` (id INT) USING delta", location);
+    S3CredFileSystem.credentialCheckEnabled = true;
+
+    // Create the catalog table referring to external table.
+    sql("CREATE TABLE %s USING delta LOCATION '%s'", TABLE_NAME, location);
+
+    // Insert 1 row into the table.
+    sql("INSERT INTO %s VALUES (1)", TABLE_NAME);
+
+    // Generate a table level hadoop configuration, with setting the delta table's all properties.
+    SerializableConfiguration serialConf =
+        new SerializableConfiguration(
+            DeltaTable.forName(session, TABLE_NAME).deltaLog().newDeltaHadoopConf());
+
+    // This Spark job consists of three main steps:
+    // 1. Read RDD to spawn Spark tasks.
+    // 2. Simulate filesystem access using the table-level Hadoop configuration to verify that
+    //    filesystem credentials are renewed the expected number of times.
+    // 3. Collect the credential renewal count.
+    //
+    // The core logic resides in the mapFunction. We use the Delta table’s Hadoop configuration
+    // to initialize the filesystem, simulating Delta table operations within a Spark executor.
+    // This allows us to accurately track how many times credentials are renewed within a task.
+    //
+    // It is possible for a Spark job to create multiple independent filesystem instances,
+    // which may misleadingly appear to renew credentials correctly even when they do not.
+    // We adopt this simulation approach because directly accessing a Spark task’s internal
+    // filesystem instance to measure credential renewals is not feasible.
+    List<Row> rows =
+        session
+            .read()
+            .format("delta")
+            .table(TABLE_NAME)
+            .toJavaRDD()
```

Comment:
if the goal is to test FS operations, then why do you need to create a table? you can just a create a RDD/task using `spark.parallelize(...)`. it will be much faster that initializing a whole table.

Reply:
> ohh .i see .. you need the table to go through the UCSingleCatalog that configures the necessary creds + disables

Yes, that's true.  I need the delta table's hadoop configuration (with the correctly configured credential properties , so that we can do the credential renewal ).

Reply:
Got it! thanks for explaining.